VSCode/armsnippets.json

```diff
@@ -506,6 +506,123 @@
     ],
     "description": "Availability Set"
 },
+"Azure Firewall": {
+    "prefix": "arm-firewall",
+    "body": [
+        "{",
+        "    \"name\": \"${1:myAzureFirewall}\",",
+        "    \"type\": \"Microsoft.Network/azureFirewalls\",",
+        "    \"apiVersion\": \"2019-04-01\",",
+        "    \"location\": \"[resourceGroup().location]\",",
+        "    \"tags\": {},",
+        "    \"properties\": {",
+        "       \"applicationRuleCollections\": [",
+        "           {",
+        "               \"name\": \"${1:appCollectionName}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${2:rulePriority:100-65000}\",",
```

Comment:
Many of these token defaults are not valid values. Supplying valid defaults (not descriptive text) will help users be successful without needing to fix all the token values when they insert a snippet. For this particular token a valid default would look like:
\"priority\": \"${2:100}\",",

Reply:
OK, I get it, but as an end user how do I know what value to provide? Either way I am likely going to have to go somewhere else to figure it out. If I take a default value in a snippet it may or may not successfully deploy (especially if it is something like a rule priority as it may conflict).

Reply:
There are a lot of different properties here, and I can add defaults so the snippet deploys with no changes. Is that the goal?

Reply:
Yes exactly - that's the goal (but it may not be possible for some properties, like resource IDs)